docs/BROKER_SPEC.md

```diff
@@ -0,0 +1,109 @@
+# Broker Specification (v0.1.0)
+
+## Context
+
+The Singer specification is pretty mature and allows a lot of 
+end-users to extract and load data from one source to a 
+specific target very simply, and allows developers to easily
+implement new taps without worrying about targets and vice-versa.
+
+However, it is not straightforward, as of now, to run Singer channels (taps + targets)
+in a production environment.
+Because logs and metrics, by default when using the [singer-python](https://github.com/singer-io/singer-python) lib, 
+are send to [stderr](https://github.com/singer-io/singer-python/blob/master/singer/logging.conf#L12), it is very difficult
+and not straightforward to get to them and exploit them.
+
+To bring the specification one step further, we need to improve on this point.
+One possible implementation to resolve this problem, would be the implementation of a *broker* type and *LOG* messages into the specification.
+
+## Definition
+
+### LOG Message
+
+A LOG message contain useful information relative to the run of the tap/broker/target.
+
+
+A LOG message have the following properties:
+
+- `level` **Required**. The level of the LOG message. Should be one of `CRITICAL`, `ERROR`, `WARNING`, `INFO`, `DEBUG`
+- `value` **Required**. The JSON formatted log
+
+The semantics of a LOG value are not part of the specification,
+and should be determined independently by each tap/broker/target.
+
+
+Example
+```
+{"type": "LOG", "type": "DEBUG", "value": {"message": "All is good there"}}
```

Comment:
I think you mean to use `level` here for debug

Reply:
You think right ;)